pod/perlre.pod

```diff
@@ -1990,6 +1990,18 @@ keep track of the number of nested parentheses. For example:
   /the (\S+)(?{ $color = $^N }) (\S+)(?{ $animal = $^N })/i;
   print "color = $color, animal = $animal\n";
 
+The use of this construct disables some optimisations globally in the
+pattern, and the pattern may execute much slower as a consequence.
+Use a double block to create an optimistic form of this construct.
+C<(?{{ ... }})> should not disable any optimisations.
+
+=item C<(?{{ I<code> }})>
+X<(?{{}})> X<regex, optimistic code>
+
+This is *exactly* the same as C<(?{ I<code> })> with the exception
+that it does not disable any optimisations in the regex engine. It
+may be even more volatile than the normal form. Introduced in 5.37.7
+it is backwards compatible with older perls where it is a no-op.
```

Comment:
Note quite exactly the same:
```
tony@venus:.../git/perl6$ ./perl -le '"aaa" =~ /(?:.(?{ local $x = $x+1; print $x }))+/'
1
2
3
tony@venus:.../git/perl6$ ./perl -le '"aaa" =~ /(?:.(?{{ local $x = $x+1; print $x }}))+/'
1
1
1
```
But changing this would be backward incompatible, since local in `(?:{})` works like this historically.

Reply:
BAH. That sucks. I had forgotten about that.  Back to the drawing board on that one.